### ISSUE TITLE
AF field is lost after decomposition of multi-allelic sites

### ISSUE THREAD

**Issue by mbootwalla** [atks/vt#73]
Hi,

After decomposition of multi-allelic sites in a vcf file the AF field in the decomposed file is missing. Here is an example:

This is an entry from the original gnomAD vcf with the VEP consequence and the HIST fields removed:
1       1599812 rs12729295      C       T,A,G   113814492.03    InbreedingCoeff AC=100865,1,1;AF=4.65571e-01,4.61578e-06,4.61578e-06;AN=216648;BaseQRankSum=-6.40000e-02;ClippingRankSum=0.00000e+00;DB;DP=8973537;FS=5.56600e+00;InbreedingCoeff=-6.27800e-01;MQ=2.92800e+01;MQRankSum=1.71000e+00;QD=1.41400e+01;ReadPosRankSum=1.35000e-01;SOR=1.25000e+00;AC_AFR=5770,0,0;AC_AMR=14997,0,1;AC_ASJ=4083,0,0;AC_EAS=6432,0,0;AC_FIN=8770,0,0;AC_NFE=45234,0,0;AC_OTH=2340,0,0;AC_SAS=13239,1,0;AC_Male=55381,1,1;AC_Female=45484,0,0;AN_AFR=12750;AN_AMR=30136;AN_ASJ=9156;AN_EAS=15504;AN_FIN=18780;AN_NFE=97614;AN_OTH=4968;AN_SAS=27740;AN_Male=118948;AN_Female=97700;AF_AFR=4.52549e-01,0.00000e+00,0.00000e+00;AF_AMR=4.97644e-01,0.00000e+00,3.31829e-05;AF_ASJ=4.45937e-01,0.00000e+00,0.00000e+00;AF_EAS=4.14861e-01,0.00000e+00,0.00000e+00;AF_FIN=4.66986e-01,0.00000e+00,0.00000e+00;AF_NFE=4.63397e-01,0.00000e+00,0.00000e+00;AF_OTH=4.71014e-01,0.00000e+00,0.00000e+00;AF_SAS=4.77253e-01,3.60490e-05,0.00000e+00;AF_Male=4.65590e-01,8.40704e-06,8.40704e-06;AF_Female=4.65548e-01,0.00000e+00,0.00000e+00;GC_AFR=1641,3698,1036,0,0,0,0,0,0,0;GC_AMR=955,13228,884,0,0,0,0,1,0,0;GC_ASJ=552,3969,57,0,0,0,0,0,0,0;GC_EAS=1888,5296,568,0,0,0,0,0,0,0;GC_FIN=835,8340,215,0,0,0,0,0,0,0;GC_NFE=4790,42800,1217,0,0,0,0,0,0,0;GC_OTH=248,2132,104,0,0,0,0,0,0,0;GC_SAS=1757,10986,1126,0,1,0,0,0,0,0;GC_Male=6873,49819,2780,0,1,0,0,1,0,0;GC_Female=5793,40630,2427,0,0,0,0,0,0,0;AC_raw=116020,1,1;AN_raw=246202;AF_raw=4.71239e-01,4.06171e-06,4.06171e-06;GC_raw=14957,100266,7876,0,1,0,0,1,0,0;GC=12666,90449,5207,0,1,0,0,1,0,0;Hom_AFR=1036,0,0;Hom_AMR=884,0,0;Hom_ASJ=57,0,0;Hom_EAS=568,0,0;Hom_FIN=215,0,0;Hom_NFE=1217,0,0;Hom_OTH=104,0,0;Hom_SAS=1126,0,0;Hom_Male=2780,0,0;Hom_Female=2427,0,0;Hom_raw=7876,0,0;Hom=5207,0,0;POPMAX=AMR,SAS,AMR;AC_POPMAX=14997,1,1;AN_POPMAX=30136,27740,30136;AF_POPMAX=4.97644e-01,3.60490e-05,3.31829e-05;DP_MEDIAN=40,36,39;DREF_MEDIAN=5.01187e-48,3.98107e-22,1.99526e-15;GQ_MEDIAN=99,99,99;AB_MEDIAN=5.00000e-01,3.05556e-01,3.33333e-01;AS_RF=6.49666e-01,2.34189e-01,1.03931e-01;AS_FilterStatus=PASS,PASS,PASS

The same variant after decomposition:
1       1599812 rs12729295      C       T       1.13814e+08     InbreedingCoeff AC=100865;AC_AFR=5770;AC_AMR=14997;AC_ASJ=4083;AC_EAS=6432;AC_FIN=8770;AC_NFE=45234;AC_OTH=2340;AC_SAS=13239;AC_Male=55381;AC_Female=45484;AN_AFR=12750;AN_AMR=30136;AN_ASJ=9156;AN_EAS=15504;AN_FIN=18780;AN_NFE=97614;AN_OTH=4968;AN_SAS=27740;AN_Male=118948;AN_Female=97700;AF_AFR=0.452549;AF_AMR=0.497644;AF_ASJ=0.445937;AF_EAS=0.414861;AF_FIN=0.466986;AF_NFE=0.463397;AF_OTH=0.471014;AF_SAS=0.477253;AF_Male=0.46559;AF_Female=0.465548;AC_raw=116020;AN_raw=246202;AF_raw=0.471239;Hom_AFR=1036;Hom_AMR=884;Hom_ASJ=57;Hom_EAS=568;Hom_FIN=215;Hom_NFE=1217;Hom_OTH=104;Hom_SAS=1126;Hom_Male=2780;Hom_Female=2427;Hom_raw=7876;Hom=5207;POPMAX=AMR;AC_POPMAX=14997;AN_POPMAX=30136;AF_POPMAX=0.497644;DP_MEDIAN=40;DREF_MEDIAN=0;GQ_MEDIAN=99;AB_MEDIAN=0.5;AS_RF=0.649666;AS_FilterStatus=PASS;OLD_MULTIALLELIC=1:1599812:C/T/A/G

1       1599812 rs12729295      C       A       1.13814e+08     InbreedingCoeff AC=1;AC_AFR=0;AC_AMR=0;AC_ASJ=0;AC_EAS=0;AC_FIN=0;AC_NFE=0;AC_OTH=0;AC_SAS=1;AC_Male=1;AC_Female=0;AN_AFR=12750;AN_AMR=30136;AN_ASJ=9156;AN_EAS=15504;AN_FIN=18780;AN_NFE=97614;AN_OTH=4968;AN_SAS=27740;AN_Male=118948;AN_Female=97700;AF_AFR=0;AF_AMR=0;AF_ASJ=0;AF_EAS=0;AF_FIN=0;AF_NFE=0;AF_OTH=0;AF_SAS=3.6049e-05;AF_Male=8.40704e-06;AF_Female=0;AC_raw=1;AN_raw=246202;AF_raw=4.06171e-06;Hom_AFR=0;Hom_AMR=0;Hom_ASJ=0;Hom_EAS=0;Hom_FIN=0;Hom_NFE=0;Hom_OTH=0;Hom_SAS=0;Hom_Male=0;Hom_Female=0;Hom_raw=0;Hom=0;POPMAX=SAS;AC_POPMAX=1;AN_POPMAX=27740;AF_POPMAX=3.6049e-05;DP_MEDIAN=36;DREF_MEDIAN=3.98107e-22;GQ_MEDIAN=99;AB_MEDIAN=0.305556;AS_RF=0.234189;AS_FilterStatus=PASS;OLD_MULTIALLELIC=1:1599812:C/T/A/G

1       1599812 rs12729295      C       G       1.13814e+08     InbreedingCoeff AC=1;AC_AFR=0;AC_AMR=1;AC_ASJ=0;AC_EAS=0;AC_FIN=0;AC_NFE=0;AC_OTH=0;AC_SAS=0;AC_Male=1;AC_Female=0;AN_AFR=12750;AN_AMR=30136;AN_ASJ=9156;AN_EAS=15504;AN_FIN=18780;AN_NFE=97614;AN_OTH=4968;AN_SAS=27740;AN_Male=118948;AN_Female=97700;AF_AFR=0;AF_AMR=3.31829e-05;AF_ASJ=0;AF_EAS=0;AF_FIN=0;AF_NFE=0;AF_OTH=0;AF_SAS=0;AF_Male=8.40704e-06;AF_Female=0;AC_raw=1;AN_raw=246202;AF_raw=4.06171e-06;Hom_AFR=0;Hom_AMR=0;Hom_ASJ=0;Hom_EAS=0;Hom_FIN=0;Hom_NFE=0;Hom_OTH=0;Hom_SAS=0;Hom_Male=0;Hom_Female=0;Hom_raw=0;Hom=0;POPMAX=AMR;AC_POPMAX=1;AN_POPMAX=30136;AF_POPMAX=3.31829e-05;DP_MEDIAN=39;DREF_MEDIAN=1.99526e-15;GQ_MEDIAN=99;AB_MEDIAN=0.333333;AS_RF=0.103931;AS_FilterStatus=PASS;OLD_MULTIALLELIC=1:1599812:C/T/A/G

As you can see above all the other multi-allelic fields were decomposed correctly except for the AF field. I ran vt decompose with the -s option. Let me know if you need any further information to help fix this.

Thanks,

Moiz

**Comment by tdelhomme** [comment]
I already had this problem and in my case the field was not declared as "A" in the header (INFO), which means as many values as possible alleles in the header. See [VCF specification](https://samtools.github.io/hts-specs/VCFv4.3.pdf).
If I'm correct, the splitting is based on this specification in the header.
Maybe you can start by verify this!

**Comment by atks** [comment]
@mbootwalla  - hi moiz, can you email the VCF file with the header and that single record to me at atks@umich.edu?

@tdelhomme - thanks for helping to troubleshoot, usually if the number of values is not specified correctly, the AF field will be retained for all decomposed variants.  So I think something else is happening here since the AF field is missing.

**Comment by tdelhomme** [comment]
Yes, sorry I understood AF fields was not splitting...

**Comment by mbootwalla** [comment]
@atks  - I apologize for taking so long to get back. I have been investigating this further and have determined that vt decompose is not the issue. The file is processed multiple times before it gets to the vt decompose step and I went back and tested the same region after every step and I believe I have found the offending piece of code. It's a regular expression that is run before piping to vt decompose and that step ends up removing the AF field only for multi-allelic variants. Since the output from this step was being piped to vt and no intermediate file was being generated I got the false impression that vt was incorrectly decomposing the variants. 

Thank you for taking the time to address my comments. I will close this issue now.
